art/estimators/classification/keras.py

```diff
@@ -373,15 +375,24 @@ def loss(self, x: np.ndarray, y: np.ndarray, reduction: str = "none", **kwargs)
 
         return loss_value
 
-    def loss_gradient(self, x: np.ndarray, y: np.ndarray, **kwargs) -> np.ndarray:
+    def loss_gradient(self, x: np.ndarray, y: np.ndarray, training_mode: bool = False, **kwargs) -> np.ndarray:
         """
         Compute the gradient of the loss function w.r.t. `x`.
 
         :param x: Sample input with shape as expected by the model.
         :param y: Target values (class labels) one-hot-encoded of shape (nb_samples, nb_classes) or indices of shape
                   (nb_samples,).
+        :param training_mode: `True` for model set to training mode and `'False` for model set to evaluation mode.
         :return: Array of gradients of the same shape as `x`.
         """
+        # pylint: disable=E0401
+        if self.is_tensorflow:
+            import tensorflow.keras.backend as k
```

Comment:
Yes, I think it's the right time for this update for learning phase.

`KerasClassifier` contains a lot of code that is necessary for earlier version of `keras` whereas since Keras 2.3 we recommend using `TensorFlowV2Classifier` for Keras models using `tf.keras` where we are not relying on `set_learning_phase`. However the name of `TensorFlowV2Classifier` can be confusing for users of Keras models. I would propose that if `KerasClassifier` is used for `keras>=2.3` that we internally replace it with `TensorFlowV2Classifier`.